### PR TITLE
[FIX] account_check_printing: enable increment in new payment

### PR DESCRIPTION
**Pull request by yosa-odoo**
Steps to reproduce:
- Create a new journal Bank
- In Sequence, find the 'New Bank Check" and edit it so the sequence can be 10 number digits long
- Create a Vendor Payments with the the New Banck and Check as a method
- Click on Print a check
- Set the check number to 2147483648 and validate
- Create a new payment with the check method
- Click on Print a check

Issue:
Error is raised

Cause:
As for https://github.com/odoo/odoo/pull/112832 there is a second check in order to increment the check number

opw-3140973

### CODE REVIEW

**Comment by yosa-odoo** [comment]
Hello @epictete,

There was still a check that needs to be corrected with a complementary flow.

I adapted the test.

File: addons/account_check_printing/tests/test_print_check.py [hunk 1]
```diff
@@ -177,3 +177,14 @@ def test_print_great_pre_number_check(self):
...
         self.assertTrue(payment.write({'check_number': '2147483647'}))
         self.assertTrue(payment.write({'check_number': '2147483648'}))
+
+        payment = self.env['account.payment'].create({
+            'payment_type': 'outbound',
+            'partner_type': 'supplier',
+            'amount': 100.0,
+            'journal_id': self.company_data['default_journal_bank'].id,
+            'payment_method_id': self.payment_method_check.id,
+        })
```
Comment (epictete) [inline addons/account_check_printing/tests/test_print_check.py:187]:
```suggestion
        vals = {
            'payment_type': 'outbound',
            'partner_type': 'supplier',
            'amount': 100.0,
            'journal_id': self.company_data['default_journal_bank'].id,
            'payment_method_id': self.payment_method_check.id,
        }
        payment = self.env['account.payment'].create(vals)
```
Maybe put the the payment values in a variable, so you can reuse them :wink:

Comment (yosa-odoo) [reply addons/account_check_printing/tests/test_print_check.py:187]: Done :)

**Comment by epictete** [comment]
Thanks @yosa-odoo :ok_hand: 
@robodoo r+

**Pull request closed without merging**